### ISSUE TITLE
Create json schemas describing both data models

### ISSUE THREAD

**Issue by areleu** [OpenEnergyPlatform/oedatamodel#56]
# Overview 
Hello, I have been working on integrating this data model without friction into frictionless. This requires effort in different repositories Here are the pull requests and issues:

Frictionless:
[https://github.com/frictionlessdata/frictionless-py/pull/1229](https://github.com/frictionlessdata/frictionless-py/pull/1229)
[https://github.com/frictionlessdata/frictionless-py/pull/1228](https://github.com/frictionlessdata/frictionless-py/pull/1228)

oemetadata:

[https://github.com/OpenEnergyPlatform/oemetadata/issues/91](https://github.com/OpenEnergyPlatform/oemetadata/issues/91)

If these changes take place then validating the models will work directly with frictionless. I think the next step is being able to validate the structure of the tables themselves, to do this we would need descriptors in json format.

# Task

Write a data descriptors for the data model. These would comprise of the json schema representation of the packages:

I can contribute to this next month but if someone starts this that would be great.

**Comment by jh-RLI** [comment]
@areleu 
I have updated the [json schema in the oemetadata](https://github.com/OpenEnergyPlatform/oemetadata/commits/develop/metadata/latest/schema.json) repo. There were some wrong entries :/. Now the validation is running successfully. Is this what you meant by this issue? 
Regarding https://github.com/OpenEnergyPlatform/oemetadata/issues/91, I will also leave this open for now, but I think it is no longer necessary (did I understand that correctly?)

**Comment by areleu** [comment]
> @areleu I have updated the [json schema in the oemetadata](https://github.com/OpenEnergyPlatform/oemetadata/commits/develop/metadata/latest/schema.json) repo. There were some wrong entries :/. Now the validation is running successfully. Is this what you meant by this issue? Regarding [OpenEnergyPlatform/oemetadata#91](https://github.com/OpenEnergyPlatform/oemetadata/issues/91), I will also leave this open for now, but I think it is no longer necessary (did I understand that correctly?)

i think https://github.com/OpenEnergyPlatform/oemetadata/issues/91 is no longer relevant. I think is better if we have a schema specific to the packages. I Started to write one but I am not done. Let me upload what I have as a github gist

**Comment by areleu** [comment]
> > @areleu I have updated the [json schema in the oemetadata](https://github.com/OpenEnergyPlatform/oemetadata/commits/develop/metadata/latest/schema.json) repo. There were some wrong entries :/. Now the validation is running successfully. Is this what you meant by this issue? Regarding [OpenEnergyPlatform/oemetadata#91](https://github.com/OpenEnergyPlatform/oemetadata/issues/91), I will also leave this open for now, but I think it is no longer necessary (did I understand that correctly?)
> 
> It is correct, i think [OpenEnergyPlatform/oemetadata#91](https://github.com/OpenEnergyPlatform/oemetadata/issues/91) is no longer relevant. I think is better if we have a schema specific to the packages. I Started to write one but I am not done. Let me upload what I have as a github gist

https://gist.github.com/areleu/6fe09323f26ec3ed4d83bc9084ab7ea9

As I said, this is not yet completely functional.

**Comment by areleu** [comment]
The justification for having such a thing is that. You can validate if the metadata complies with the OEMetadata but you can't really check if it is a valid OEdatamodel implementation. For doing so one would need to write a schema that extends the OEMetadata to check for the particular constraints of the OEDatamodel. 

This way you can validate very easily if an user is filling their tables correctly.

If we have these schemas, I think we can even use frictionless for validation out of the box.

**Comment by jh-RLI** [comment]
@henhuy you mentioned some updates to the oedatamodel. @areleu is working on a json schema that can be used with frionless to validate data. Maybe it is useful to share the changes so we can also create a schema for the updated version.

**Comment by jh-RLI** [comment]
@henhuy I think you have already implemented this in the oeadatamodel api?

**Comment by henhuy** [comment]
Yes. I implemented an API for our project, which loads metadata from existing OEP table, creates frictionless schema from OEMetadata schema and validates incoming data from CSV file via frictionless resource. If everything is okay, data gets uploaded otherwise a frictionless error report is returned to user.
Here the corresponding code for [loading and validating data via OEMetadata](https://github.com/open-modex/oedatamodel_api/blob/feature/validate_upload_data/oedatamodel_api/webapp.py#L290) (you have to look into upload module)
